### PR TITLE
Added typescript definitions

### PR DESCRIPTION
**Pull request by orendin**
- we are using the library in our typescript apps but always have to include our own, separate typescript defintion file.
- now including this into the library itself and created a pull request
- it would be helpful for other people using this with typescript if you could merge this into main/master.

### CODE REVIEW

**Comment by bsyk** [comment]
@orendin this is great, thanks for contributing. The returned token object also includes a 'renew_time' property. While most consumers don't use that, would it cause any issue if a typescript consumer wanted to use it and it not be included in the IToken definition?

File: predix-uaa-client.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+declare module "predix-uaa-client" {
+    export function getToken(url: string, clientId: string, clientSecret: string, refreshToken?: string): Promise<IToken>;
+
+    export function clearCache();
+
+    export interface IToken {
+        access_token: string;
+        refresh_token: string;
+        expire_time: string;
```
Comment (bsyk) [inline predix-uaa-client.d.ts]: `expire_time` Is a number (ms since epoch).

**Comment by orendin** [comment]
You are right on both terms. I have updated the code accordingly and committed the changes.

**Comment by bsyk** [comment]
I'll update the version number and publish to npm tomorrow.

**Pull request merged**